pkg/ingester/client/cortex.proto

```diff
@@ -34,6 +34,7 @@ message WriteRequest {
     RULE = 1;
   }
   SourceEnum Source = 2;
+  repeated MetricMetadata metadata = 3 [(gogoproto.nullable) = false, (gogoproto.customtype) = "PreallocMetricMetadata"];
```

Comment:
Given we do expect a low traffic on metadata, I've the feeling using `PreallocMetricMetadata` is just an overcomplication (and premature optimization). Have you considered using no custom type and thus no struct recycling?

Reply:
I did. Given my experience with struct recycling is very limited (and I have no data to understand how big of an impact this might be on Cortex), I had no idea how relevant was this optimisation. I wanted to make sure that this work wouldn't make the write path _much_ worse.

I'm happy to trust your judgement here and if you think it's not needed - I'll remove it.